src/test/java/com/stripe/functional/StripeClientTest.java

```diff
@@ -0,0 +1,49 @@
+package com.stripe.functional;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.stripe.BaseStripeTest;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class StripeClientTest extends BaseStripeTest {
+
+  private Boolean originalTelemetry;
+
+  @BeforeEach
+  public void setUp() {
+    this.originalTelemetry = Stripe.enableTelemetry;
+    Stripe.enableTelemetry = true;
+  }
+
+  @AfterEach
+  public void tearDown() {
+    Stripe.enableTelemetry = originalTelemetry;
+  }
+
+  @Test
+  public void testReportsUsageTelemetry() throws StripeException {
+    mockClient.customers().create();
+    Mockito.reset(httpClientSpy);
+    mockClient.customers().update("cus_xyz");
+    Mockito.verify(httpClientSpy)
```

Comment:
Do we not have a cute wrapper around httpClientSpy to make these things easier?

Reply:
BaseStripeTest's existing wrappers operate on `networkSpy` which uses a mock `ResponseGetter`. I wanted this test to be a little closer to the metal to exercise more of the changes in this PR, so it uses `httpClientSpy`, which didn't have any wrappers -- but I added a small wrapper in [Improve test infra](https://github.com/stripe/stripe-java/pull/1721/commits/43d6ac95304a35384a40aec78f7de42a64259924)